### PR TITLE
Add server status code tests; update Student test snapshot

### PR DESCRIPTION
**Pull request by aqchin**
Additionally:

- Set CircleCI to deploy PR to staging repo instead of production
- Add catch blocks to server requests (namely req with /getParticular*)

**Important note**: It looks like pull requests are only being ran through CircleCI when made by myself but not others. I will be looking into this problem.

### CODE REVIEW

**Comment by BeardOfDan** [comment]
At any point did you set up an authentication key or something for your GitHub? If so, then we likely all have to do it, or we might be able to get away with giving it to our organization.

File: server.js [hunk 1]
```diff
@@ -103,7 +103,10 @@ app.post('/getParticularStudent', (req, res, next) => {
   db.loadParticular('student', req.body.studentInfo)
     .then((data) => {
       res.send(JSON.stringify(data));
-    })
+
```
Comment (philcho) [inline server.js:106]: Yeah, I guess we were being too optimistic :)

File: src/test/Student.test.js [hunk 2]
```diff
@@ -9,9 +9,10 @@ Enzyme.configure({ adapter: new Adapter() });
 
 describe('Student', () => {
   it('should render correctly', () => {
+    const studentData = {};
     const output = shallow(
-      <Student />
+      <Student studentData={studentData} />
```
Comment (philcho) [inline src/test/Student.test.js:14]: Nice. I'll follow up on this with a version that uses mock data.

**Comment by aqchin** [comment]
I granted it access to the org through my account. I think the issue is that I did not set it to build from forked pull requests. I have to wait for another PR from someone else to see if it works. If you'd like, you can sign in to Circle with your Github to view the settings for the project.

File: test/server.test.js [hunk 3]
```diff
@@ -4,14 +4,179 @@ import request from 'supertest';
...
-  it('should respond with status 200 when GET is sent to /template', (done) => {
-    request(app)
-      .get('/template')
-      .expect(200)
-      .end((err, res) => {
-        if (err) return done(err);
-        done();
-      });
-  }); 
+  describe('Template path', () => {
```
Comment (philcho) [inline test/server.test.js]: I wonder if we still need the template routes and files. We can chat about this later.

Comment (aqchin) [reply test/server.test.js]: Yeah, I should probably remove it

File: test/server.test.js [hunk 4]
```diff
@@ -4,14 +4,179 @@ import request from 'supertest';
...
+      request(app)
+        .post('/getParticularShoutout')
+        .expect(200)
+        .end((err, res) => {
+          if (err) return done(err);
+          done();
+        });
+    }); 
+
+    it('should respond with status 200 when POST is sent to /getParticularShoutout', (done) => {
```
Comment (philcho) [inline test/server.test.js]: Do we need both this test and the one just before it? (which doesn't include the req body data)

Comment (aqchin) [reply test/server.test.js]: Probably not, I changed it because it wasn't sending a 404 but I'll update the PR to remove it

**Review by philcho — APPROVED** [review]

**Pull request merged**